store/storetest/channel_store.go

```diff
@@ -8415,3 +8416,82 @@ func testDeleteSidebarCategory(t *testing.T, ss store.Store, s SqlSupplier) {
 		assert.NotNil(t, err)
 	})
 }
+
+func testGetDirectChannelsForUser(t *testing.T, ss store.Store) {
+	teamID := model.NewId()
+
+	c1 := model.Channel{}
+	c1.TeamId = teamID
+	c1.DisplayName = "Different Name" + model.NewId()
+	c1.Name = "zz" + model.NewId() + "b"
+	c1.Type = model.CHANNEL_DIRECT
+
+	c2 := model.Channel{}
+	c2.TeamId = teamID
+	c2.DisplayName = "Different Name 1" + model.NewId()
+	c2.Name = "zz1" + model.NewId() + "c"
+	c2.Type = model.CHANNEL_DIRECT
+
+	t.Run("direct messages with created user & others", func(t *testing.T) {
```

Comment:
Not sure if a single sub-test makes sense here. If you want, we can split the cases into u1 and then u2 and u3 together. Otherwise, we can just have a single test.